src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -0,0 +1,152 @@
+    }
+    return views;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    List<String> fileNames = new ArrayList<>();
+    fileNames.add(TEST_TEMPLATE_FILE);
+    // TODO(shinfan): Add gRPC service mock template.
+    return fileNames;
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new JavaTypeTable(), new JavaModelTypeNameConverter());
+  }
+
+  private void addImports(SurfaceTransformerContext context) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.saveNicknameFor("org.junit.After");
+    typeTable.saveNicknameFor("org.junit.AfterClass");
+    typeTable.saveNicknameFor("org.junit.Before");
+    typeTable.saveNicknameFor("org.junit.BeforeClass");
+    typeTable.saveNicknameFor("org.junit.Test");
+    typeTable.saveNicknameFor("java.io.IOException");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.ValueGenerator");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.LocalServiceHelper");
+    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
+    typeTable.saveNicknameFor("junit.framework.Assert");
+  }
+
+  private GapicSurfaceTestClassView createTestClassView(
+      Interface service, SurfaceTransformerContext context, ApiConfig apiConfig) {
+    String outputPath = pathMapper.getOutputPath(service, apiConfig);
+
+    GapicSurfaceTestClassView testClass =
+        GapicSurfaceTestClassView.newBuilder()
+            .packageName(context.getApiConfig().getPackageName())
+            .apiSettingsClassName(context.getNamer().getApiSettingsClassName(service))
+            .apiClassName(context.getNamer().getApiWrapperClassName(service))
+            .name(context.getNamer().getTestClassName(service))
+            .mockServiceClassName(context.getNamer().getMockServiceClassName(service))
+            .testCases(createTestCaseViews(service, context, apiConfig))
+            .outputPath(outputPath)
+            .imports(context.getTypeTable().getImports())
+            .templateFileName(TEST_TEMPLATE_FILE)
+            .build();
+    return testClass;
+  }
+
+  private List<GapicSurfaceTestCaseView> createTestCaseViews(
+      Interface service, SurfaceTransformerContext context, ApiConfig apiConfig) {
+    ArrayList<GapicSurfaceTestCaseView> testCaseViews = new ArrayList<>();
+    for (Method method : service.getMethods()) {
+      testCaseViews.add(createTestCaseView(method, service, context, apiConfig));
+    }
+    return testCaseViews;
+  }
+
+  private GapicSurfaceTestCaseView createTestCaseView(
+      Method method, Interface service, SurfaceTransformerContext context, ApiConfig apiConfig) {
+    MethodConfig methodConfig = apiConfig.getInterfaceConfig(service).getMethodConfig(method);
+    MethodTransformerContext methodContext =
+        MethodTransformerContext.create(
+            service, apiConfig, context.getTypeTable(), context.getNamer(), method, methodConfig);
```

Comment:
I think it would be better to create the MethodTransformerContext in the calling function; simply use .asMethodContext() on SurfaceTransformerContext.